native/src/components/ExportEventButton.tsx

```diff
@@ -27,45 +28,57 @@ const ExportEventButton = ({ event }: ExportEventButtonType): ReactElement => {
-    } else if (Platform.OS === 'android') {
-      Linking.openURL(`content://com.android.calendar/events/${id}`)
+  const exportEventToCalendar = async (calendarId: string | undefined): Promise<void> => {
+    let startDate = event.date.startDate.toUTC().toString()
+    let endDate = event.date.endDate.toUTC().toString()
+    const allDay = event.date.allDay
+    if (Platform.OS === 'android' && allDay) {
+      // If allDay is set to true, Android demands that the time has a midnight boundary.
+      // The endDate we receive from the CMS for allDay events is always at 23:59:00.
+      startDate = event.date.startDate.toFormat("yyyy-LL-dd'T'00:00:00.000'Z'")
+      endDate = event.date.endDate.plus({ minutes: 1 }).toFormat("yyyy-LL-dd'T'00:00:00.000'Z'")
     }
-  }
 
-  const exportEventToCalendar = async (calendarId: string | undefined): Promise<void> => {
     try {
-      const id = await RNCalendarEvents.saveEvent(event.title, {
-        startDate: event.date.startDate.toISO(),
-        endDate: event.date.endDate.toISO(),
-        allDay: event.date.allDay,
-        location: event.location?.fullAddress,
-        description: event.excerpt,
+      await RNCalendarEvents.saveEvent(event.title, {
+        startDate,
+        endDate,
+        allDay,
         calendarId,
+        location: event.location?.fullAddress,
+        description: event.excerpt, // Android
+        notes: event.excerpt, // iOS
       })
       setEventExported(true)
       showSnackbar({
         text: t('added'),
-        positiveAction: {
-          label: t('goToCalendar'),
-          onPress: () => openCalendarApp(event, id),
-        },
       })
-    } catch {
+    } catch (e) {
       showSnackbar({ text: 'generalError' })
+      reportError(e)
     }
   }
 
+  const checkPermissionState = (
+    permissions:
+      | PermissionStatus
+      | Record<'android.permission.READ_CALENDAR' | 'android.permission.WRITE_CALENDAR', PermissionStatus>,
+    state: PermissionStatus,
+  ): boolean => {
+    if (typeof permissions === 'string') {
+      return permissions === state
+    }
+    return (
+      permissions['android.permission.READ_CALENDAR'] === state &&
+      permissions['android.permission.WRITE_CALENDAR'] === state
+    )
+  }
+
   const checkCalendarsAndExportEvent = async (): Promise<void> => {
-    const checkedPermission = await RNCalendarEvents.checkPermissions()
-    const permission =
-      checkedPermission === 'undetermined' ? await RNCalendarEvents.requestPermissions() : checkedPermission
-    if (permission === 'denied' || permission === 'restricted') {
+    const iosPermission = PERMISSIONS.IOS.CALENDARS
+    const androidPermissions = [PERMISSIONS.ANDROID.READ_CALENDAR, PERMISSIONS.ANDROID.WRITE_CALENDAR]
+    const permission = Platform.OS === 'ios' ? await request(iosPermission) : await requestMultiple(androidPermissions)
+    if (checkPermissionState(permission, 'limited') || checkPermissionState(permission, 'blocked')) {
```

Comment:
Optional: If you use an array for iOS as well you can simplify it and the delete the function above:
```suggestion
    const iosPermission = [PERMISSIONS.IOS.CALENDARS]
    const androidPermissions = [PERMISSIONS.ANDROID.READ_CALENDAR, PERMISSIONS.ANDROID.WRITE_CALENDAR]
    const permissions = await requestMultiple(Platform.OS === 'ios' ? iosPermission : androidPermissions)
    const permissionsDenied = Object.values(permissions).some(permission =>
      ['limited', 'blocked'].includes(permission),
    )
       
    if (permissionsDenied) {
```

Reply:
Great idea, thanks!